frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,892 @@
+		///
+		/// The dispatch origin for this call must be master curator.
+		/// parent bounty must be in "active" state.
+		///
+		/// Child-bounty gets added successfully & fund gets transferred from
+		/// parent bounty to child-bounty account, if parent bounty has
+		/// enough fund. Else call gets failed.
+		///
+		/// Upper bound to maximum number of active  child-bounties that
+		/// can be added are managed via runtime trait config
+		/// 'MaxActiveChildBountyCount'.
+		///
+		/// If the call is success, the state of child-bounty is
+		/// moved to "Added" state.
+		///
+		/// - `bounty_id`: Bounty ID for which child-bounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the child-bounty.
+		#[pallet::weight(<T as Config>::WeightInfo::add_child_bounty(description.len() as u32))]
+		pub fn add_child_bounty(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			#[pallet::compact] value: BalanceOf<T>,
+			description: Vec<u8>,
+		) -> DispatchResult {
+			let signer = ensure_signed(origin)?;
+
+			// Verify the arguments.
+			let bounded_description =
+				description.try_into().map_err(|_| BountiesError::<T>::ReasonTooBig)?;
+			ensure!(value >= T::ChildBountyValueMinimum::get(), BountiesError::<T>::InvalidValue);
+			ensure!(
+				Self::parent_child_bounties(parent_bounty_id) <=
+					T::MaxActiveChildBountyCount::get() as u32,
+				Error::<T>::TooManyChildBounties,
+			);
+
+			let (curator, _) = Self::ensure_bounty_active(parent_bounty_id)?;
+			ensure!(signer == curator, BountiesError::<T>::RequireCurator);
+
+			// Read parent bounty account info.
+			let parent_bounty_account =
+				pallet_bounties::Pallet::<T>::bounty_account_id(parent_bounty_id);
+
+			// Ensure parent bounty has enough balance after adding child-bounty.
+			let balance = T::Currency::free_balance(&parent_bounty_account);
+			ensure!(
+				balance.saturating_sub(value) >= T::Currency::minimum_balance(),
+				Error::<T>::InsufficientBountyBalance,
+			);
+
+			// Create child-bounty ID.
+			let child_bounty_id = Self::child_bounty_count();
+
+			// Transfer fund from parent bounty to child-bounty.
+			let child_bounty_account = Self::child_bounty_account_id(child_bounty_id);
+			T::Currency::transfer(
+				&parent_bounty_account,
+				&child_bounty_account,
+				value,
+				AllowDeath,
+			)?;
+
+			// Increment the active child-bounty count.
+			<ParentChildBounties<T>>::mutate(parent_bounty_id, |count| {
+				*count = count.saturating_add(1)
+			});
+			<ChildBountyCount<T>>::put(child_bounty_id.saturating_add(1));
+
+			// Create child-bounty instance
+			Self::create_child_bounty(parent_bounty_id, child_bounty_id, bounded_description);
+			Ok(())
+		}
+
+		/// Propose curator for funded child-bounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		///
+		/// Parent bounty must be in active state,
+		/// for this child-bounty call to work.
```

Comment:
line is wrapping too short.

(can recommend: https://marketplace.visualstudio.com/items?itemName=stkb.rewrap)